src/Microsoft.DotNet.Build.Tasks/GenerateBindingRedirect.cs

```diff
@@ -25,25 +27,66 @@ public override bool Execute()
         {
             ns = "urn:schemas-microsoft-com:asm.v1";
             XElement bindingRedirectAssemblies = new XElement(ns + "assemblyBinding");
-            foreach (ITaskItem assembly in AssemblyIdentities)
+            foreach (ITaskItem assembly in Assemblies)
             {
-                string publicKeyToken = assembly.GetMetadata("PublicKeyToken");
-                string assemblyVersion = assembly.GetMetadata("Version");
-                string assemblyName = assembly.GetMetadata("Name");
+                AssemblyName result = null;
+                try
+                {
+                    using (var assemblyStream = new FileStream(assembly.ItemSpec, FileMode.Open, FileAccess.Read, FileShare.Delete | FileShare.Read))
```

Comment:
Consider keeping consistent usage of `var` within this file, which would mean not using it here.